api/v1beta1/heat_webhook.go

```diff
@@ -238,3 +238,29 @@ func (r *Heat) ValidateDelete() (admission.Warnings, error) {
 	// TODO(user): fill in your validation logic upon object deletion.
 	return nil, nil
 }
+
+// SetDefaultRouteAnnotations sets HAProxy timeout values of the route
+func (spec *HeatSpecCore) SetDefaultRouteAnnotations(annotations map[string]string) {
+	const haProxyAnno = "haproxy.router.openshift.io/timeout"
+	// Use a custom annotation to flag when the operator has set the default HAProxy timeout
+	// With the annotation func determines when to overwrite existing HAProxy timeout with the APITimeout
+	const heatAnno = "api.heat.openstack.org/timeout"
+
+	valHeat, okHeat := annotations[heatAnno]
+	valHAProxy, okHAProxy := annotations[haProxyAnno]
+
+	// Human operator set the HAProxy timeout manually
+	if !okHeat && okHAProxy {
+		return
+	}
+
+	// Human operator modified the HAProxy timeout manually without removing the Heat flag
+	if okHeat && okHAProxy && valHeat != valHAProxy {
+		delete(annotations, heatAnno)
+		return
+	}
+
+	timeout := fmt.Sprintf("%ds", spec.APITimeout)
+	annotations[heatAnno] = timeout
+	annotations[haProxyAnno] = timeout
```

Comment:
How are these being used? This function isn't called from anywhere that I can see, and it doesn't return anything. Should we be taking in a pointer to an annotations map instead?

Reply:
I'm following the pattern of calling it from openstack-operator webhook:
- https://github.com/openstack-k8s-operators/manila-operator/blob/main/api/v1beta1/manila_webhook.go#L241
- https://github.com/openstack-k8s-operators/openstack-operator/blob/main/apis/core/v1beta1/openstackcontrolplane_webhook.go#L783
- https://github.com/search?q=org%3Aopenstack-k8s-operators%20SetDefaultRouteAnnotations&type=code

after this PR gets merged, I'll open a PR against openstack-operator

Reply:
https://github.com/openstack-k8s-operators/openstack-operator/pull/1163